tiiuae/ghaf
# graphics: add greetd login manager

Adds a login manager and a graphical greeter in Ghaf, with option for autologin (by default for now).

This removes redundant services such as ghaf-session, and seatd (as replaced with an official NixOS service option).

## Description of changes

- Adds greetd login manager
  - This runs as a separate user 'greeter' which is in 'video' group.
  - Remove `ghaf-session` service, as sessions are managed by greetd.
- Use gtkgreet as a greeter, customised to match our style guide.
  - We may switch to something more customisable, such as ReGreet.
- Uses labwc compositor to run the greeter
  - Brightness controls, alternative keyboard layouts works on the login screen.
  - Not possible to launch Terminal from greeter user.
  - labwc supports multi-monitor well, as well as `wlr-layer-shell-unstable` for gtkgreet.
- Replace custom seatd service with `services.seatd` as it is now available in NixOS.
- Set wlroots environment variables using `environment.sessionVariables`.
  - This allows other compositor to work properly, but this PR doesn't add any new compositors.
- Fix renderer option for Orin (previously it seemed to fallback correctly even with typo).


I have initially tested greeter with cage. Cage doesn't support `wlr-layer-shell-unstable`, which causes gtkgreet to only show on one screen  Also Cage doesn't seem to set the resolution correctly on the Orin, and generally has crude support for multi-monitors. These issues are resolved when using labwc.

## Checklist for things done





## Testing

- By default, you will be logged in automatically.
- You can log out by clicking Log Out in app drawer, which should bring you to the login screen.
- Setting `ghaf.graphics.labwc.autologinUser = null;` would disable autologin, and show the login screen on boot.
- Login, logout multiple times. Should not cause system to go in bad state.
- In login screen, plug and unplug external monitors. Should not glitch or crash.
  - Login prompt is shown on monitor with cursor, moving cursor to other monitor would move the prompt.
  - Other monitors should only show clock on the centre.
- All applications should be launchable after re-logins.
- Boot time shouldn't be significantly affected.
- Fingerprint authentication should work with both lock screen and login screen. Login screen should not hog access to fingerprint reader after login.
  - With same previous caveats with gtklock.
